example/lib/main.dart

```diff
@@ -30,14 +35,58 @@ class _MyAppState extends State<MyApp> with SingleTickerProviderStateMixin {
   void initState() {
     super.initState();
     controller = TabController(length: 3, vsync: this);
-
+    initPlatformState();
     addFlutterTag();
   }
 
   static void addFlutterTag() {
     Airship.addTags(["flutter"]);
   }
 
+// Platform messages are asynchronous, so we initialize in an async method.
+  Future<void> initPlatformState() async {
+    Airship.onPushReceived.listen((event) {
+      debugPrint('Push Received $event');
+    });
+
+    Airship.onNotificationResponse
+        .listen((event) => debugPrint('Notification Response $event'));
+
+    Airship.onDeepLink.listen((event){
+      const home_tab =  0;
+      const message_tab =  1;
+      const settings_tab =  2;
+
+      switch(event) {
+        case home_deep_link: {
+          controller.animateTo(home_tab);
+          break;
+        }
+        case message_center_deep_link: {
+          controller.animateTo(message_tab);
+          break;
+        }
+        case settings_deep_link: {
+          controller.animateTo(settings_tab);
+          break;
+        }
+      }
+    });
+
+    Airship.onInboxUpdated
+        .listen((event) => debugPrint('Inbox updated link'));
+
+    Airship.onShowInbox
+        .listen((event) => debugPrint('Show inbox'));
+
+    Airship.onShowInboxMessage
+        .listen((messageId) => debugPrint('Show inbox message $messageId'));
+
+    Airship.onChannelRegistration.listen((event) {
+      debugPrint('Channel registration $event');
```

Comment:
minor formatting nit: do we want indentation to be two or four spaces?

Reply:
I copied and re-pasted this and this is what it auto-formats to. The auto formatting indents the .listen call over slightly farther. Since this is what it does automatically I'm not going to mess with it.